add latitude and longitude configuration to darksky sensor

RickyTaterSalad opened the pull request:
Allow configuration of latitude and longitude in the darksky sensor which
falls back to home assistant coordinates if latitude/longitude not
specified.
Would be nice to check for both lat and long the sensor config since you can specify a lat in the sensor and have the long read from the home automation configuration. I did not do this since the openweathermap sensor allows lat/long mix from sensor config and home assistant config.

**Description:**


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

**Example entry for `configuration.yaml` (if applicable):**
```yaml

```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

RickyTaterSalad committed:
Optional latitude and longitude to darksky sensor

allow configuration of latitude and longitude in the darksky sensor.
falls back to home assistant coordinates if latitude/longitude not
specified.

homeassistant commented:
Hi @RickyTaterSalad, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6191).

Once you do that we will be able to review and accept this pull request.

Thanks!

balloob commented on homeassistant/components/sensor/darksky.py:
```diff
@@ -117,6 +118,8 @@
     vol.Required(CONF_API_KEY): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_UNITS): vol.In(['auto', 'si', 'us', 'ca', 'uk', 'uk2']),
+    vol.Optional(CONF_LATITUDE): cv.latitude,
```
Please ensure that latitude and longitude are always set in a group like here: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/influxdb.py#L35-L36

balloob commented on homeassistant/components/sensor/darksky.py:
```diff
@@ -127,7 +130,9 @@
     """Setup the Dark Sky sensor."""
     # Validate the configuration
-    if None in (hass.config.latitude, hass.config.longitude):
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
```
This is weird. Now it's possible to use HASS latitude but use configured longitude? Please make it either.

RickyTaterSalad replied:
Will do, heads up on this.. code was taken from components\weather\openweathermap.py so this file should probably handle lat/long the same

balloob commented on homeassistant/components/sensor/darksky.py:
```diff
@@ -127,7 +130,9 @@
-    if None in (hass.config.latitude, hass.config.longitude):
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    if None in (latitude, longitude):
```
This is impossible and this check can be removed

balloob reviewed (changes requested):
Will also require tests to be added.

pvizeli commented:
Ping. Any planes to finish that?

RickyTaterSalad commented:
yeah, working on it now. My changes were just copied from other sensor code so i'm still learning the framework and libs used in this project.

RickyTaterSalad committed:
lat/long as inclusive on schema. removed None check for lat/long in setup_platorm

altered schema to require latitude and longitude configured as a pair.
removed None check on latitude and longitude since values will fall back
to hass config if not present

RickyTaterSalad committed:
adhere to line limit of 79 characters

adhere to line limit of 79 characters

balloob commented:
Sorry for the delay, looks good 🐬
